### DISCUSSION TITLE
Bluetooth With LXC?

### DISCUSSION THREAD

**Discussion by guarddog13** [tteck/Proxmox#598] [Help Needed]
I have tried this now with a couple of different BT a adapters. I am running the Home Assistant container LXC, the one with portainer. I successfully passed thru my USB devices by having it privileged. I am able to see all my USB sticks with lsusb. The HA container has access to them as the zigbee stick shows as a new integration. However the BT device doesn't show. It's not a docker/ha issue as the host lxc cannot start bluetoothctl. 

I am able to successfully pass the BT devices to VMs fine and they work there. The host node can also access it. Seems that hci0 devices don't play well with the underlying os to the lxc...any ideas here? Is this an lxc issue, a debian issue...

I wonder if HA core would be better or the Podman version.

Does anyone have any ideas getting this BT device to properly work inside the LXC?

**Comment by guarddog13** [comment]
From what I'm finding BT devices don't passthru very well due to the way the Linux kernel is. HAos passed it through fine so idk. However, from what I've read BT devices register as network devices but don't have a name space...so they can't be passed through to LXCs. I'm finding that in trying to pass it through to Ubuntu is the exact same issue. I get the exact same error in my Ubuntu VM as I do the LXCs. 

It looks like I either have to choose BT and HAos or HA in a Container without BT. I'm quite enjoying not have the Supervisor and I don't have one BT device so  🤷

**Comment by guarddog13** [comment]
@tteck what are the LXCs for HA built on? Is it Ubuntu or Debian? If it's Ubuntu any chance we can get a Debian version? Seems that Ubuntu is what is not playing nice with BT dongles. I had my AgentDVR and Deepstack in a Debian VM already and moved HA to that VM and passed the BT dongle through. Seems Debian has better support for these things. Debian in general seems like a more stable OS as is. 

Disregard if they are built on Debian because then it's a LXC issue as mentioned previously.

**Reply by tteck** [reply]
Debian 11
https://github.com/tteck/Proxmox/blob/a67a9138f69011f96d28c2e668f3a2fd35da5ac7/ct/homeassistant-v4.sh#L7

**Reply by guarddog13** [reply]
Thank you. Its an LXC issue then. I'll live with it. The only downfall is using a VM means larger, slower backups....

Oh well they happen when I'm asleep and I have more than enough storage.

**Reply by tteck** [reply]
You are using a privileged LXC, correct?

**Reply by guarddog13** [reply]
I am. According to what I've found on the proxmox forums is that due to kernel limitations BT dongles cant be passed thru to LXCs. They register as network devices with no name space...therefore unless you pass thru the dbus socket BT dongles won't pass thru. They also point out that passing thru the dbus is an enormous security risk...

**Comment by tteck** [comment]
Maybe use a bluetooth proxy?
https://esphome.github.io/bluetooth-proxies/

**Reply by tteck** [reply]
Yep, you need a Bluetooth proxy
https://shop.m5stack.com/products/atom-lite-esp32-development-kit?ref=NabuCasa

**Reply by tteck** [reply]
https://digiblur.com/2022/09/07/home-assistant-esphome-bluetooth-proxy-how-to/

**Comment by guarddog13** [comment]
That was a suggestion on the HA forums too. I don't have any devs boards or the experience with them. I don't mind having HA inside the VM. I already have the VM running as AgentDVR is terrible inside an LXC. Adding HA to it didn't add much more overhead than the lxc was using. I installed a small GUI desktop that's probably using more than HA 😆

**Reply by tteck** [reply]
I just ordered one of those m5stacks to play around with. 😉

**Reply by guarddog13** [reply]
The ibeacons are interesting but i feel that using the companion app or even own tracks with mqtt is a better presence method. I wish more things supported mqtt as it is quite literally my most reliable service. 

I have not one pure BT device. I've never found a better pure BT option for my iot devices than zwave/zigbee/wifi. 

What I'm excited for is my stuff that is wifi cloud, BT local... like my govee light strip. 

I'm also curious about something like my Viper Smart Start at least at home. Currently i have an apache server running a token getter for their api as it's locked. The device has BT capabilities so it makes me wonder if we'll see someone make a local BT option for the smartstart. Unlikely as it's popularity is nowhere near drone mobile. 


I'm curious if something with these BT proxies could be built for it though too. 

A local option and maybe more with hotspot in my truck? A local option would be amazing. When viper changed their api last it took 6 months for the internet to find a way around it. 

The bt only option would mean no subscription and if they break their api i still have control. But how do you get the bt to talk to HA away from home? Hmm a pi in the truck with its own HA? 🤦‍♂️ I have to stay off these forums. 🤣

**Reply by guarddog13** [reply]
Sorry way off topic lol

**Comment by bultodepapas** [comment]
hi, in 2024 i still having the same problem with the lxc homeassistant and bluetooth, any fix?
